### PR TITLE
Fixes #5468 - prevent Chrome from autofilling passwords

### PR DESCRIPTION
**Pull request by shlomizadok**
1) Create a dummy password field above each password field in password_f
2) Hide that field with css and clear its value.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -391,3 +391,8 @@ function setPowerState(item, status){
   power_actions.hide();
   $('[rel="twipsy"]').tooltip();
 }
+
+// Removes the value from fake password field.
+$(function(){
+  $("#fakepassword").val("")
+})
```
Comment (lzap) [inline app/assets/javascripts/application.js]: I am no JS expert but is this the right place to do it? Just confirming with others...

Going to test this.

File: app/helpers/layout_helper.rb [hunk 2]
```diff
@@ -50,6 +50,8 @@ def textarea_f(f, attr, options = {})
   end
 
   def password_f(f, attr, options = {})
+    password_field_tag(:fakepassword) +
```
Comment (lzap) [inline app/helpers/layout_helper.rb]: I think we want this to be optional, I definitely like to store the application sign-in password.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (domcleal) [inline app/assets/javascripts/application.js]: should be in onContentLoad I'd guess

**Comment by lzap** [comment]
It renders fine please make it optional and allow storing password on the main login dialog.

Then I will test this across the application. Looks good.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (shlomizadok) [inline app/assets/javascripts/application.js]: $ is attached to the dom, so this place is more solid and won't fail if there are errors on onContentLoad

**Comment by lzap** [comment]
Also leave a comment there about the Chrome bug/feature so we don't remove this in future :-)

File: app/helpers/layout_helper.rb (hunk 2 above)
Comment (shlomizadok) [inline app/helpers/layout_helper.rb]: sign in password does not use password_f (see [login.html.erb](https://github.com/theforeman/foreman/blob/develop/app/views/users/login.html.erb) ). All others do use password_f.
do you still think it should be optional?

**Comment by shlomizadok** [comment]
@lzap I think it doesn't need to be optional (see my [comment](https://github.com/theforeman/foreman/pull/1761#discussion_r17544421) )

**Comment by lzap** [comment]
Ok I get it. Will test tomorrow, thanks.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (lzap) [inline app/assets/javascripts/application.js]: I am fine with that.

**Comment by lzap** [comment]
Ok I tested this and it works fine, nothing is really stored in Chrome managed passwords. Anyone want to chip in? @domcleal ?

**Comment by lzap** [comment]
Dominic is right (on IRC) that this could possibly break other browsers. I will re-test with Firefox. Anyone else having different broswer (e.g. MacOS X one, or the one in Microsoft Windows not sure how it is called) and can test this patch please?

**Comment by lzap** [comment]
Ok this does not work with Firefox. It offers me to update password for "admin" user which is set to what I type in as root password. Can we prevent from that as well? Opinions?

**Comment by shlomizadok** [comment]
@lzap @domcleal - moved the code into `onContentLoad` (works better on ajax forms) . I have tested it on FF & Chrome and it works well.
This change removes "fakepassword" so FF won't ask to update password and it is not sent to the server.
Opinions?

**Comment by lzap** [comment]
I am afraid my FF still offers me to remember password for Admin user. Using FF 32.0.

Try to test this without any stored password, and then with stored (main login) pass.

**Comment by shlomizadok** [comment]
@lzap, When changing a password field in FF, it offers to update the password. This is a default FF behavior which we can't change. The purpose of this PR was to prevent browsers to autofill the currently saved password. @domcleal opinions?

File: app/helpers/layout_helper.rb (hunk 2 above)
Comment (ohadlevy) [inline app/helpers/layout_helper.rb]: can you add the hide css class to it instead of the extra css?

**Comment by lzap** [comment]
I am okay with this behavior. @ohadlevy ?

**Comment by dLobatog** [comment]
:+1: too, works well for me. I'm only labeling it until tests are fixed.

**Comment by shlomizadok** [comment]
[test]

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -101,6 +101,12 @@ function onContentLoad(){
   });
 
   multiSelectOnLoad();
+
+  // Removes the value from fake password field.
+  $("#fakepassword").val("");
+  $('form').on('click', 'input[type="submit"]', function() {
```
Comment (ohadlevy) [inline app/assets/javascripts/application.js:107]: in several places (e.g. the host form submit) we send the form data in ajax, how can we ensure that this happens prior to the ajax?

once strong params are used, you might get an exception

Comment (ohadlevy) [inline app/assets/javascripts/application.js:107]: ajax of course happens via another form on click

Comment (shlomizadok) [inline app/assets/javascripts/application.js:107]: click event comes before submit event, which happens on host submit.

**Comment by shlomizadok** [comment]
[test]

**Comment by shlomizadok** [comment]
@ohadlevy tests are :green_apple: - I think we can merge

**Comment by ohadlevy** [comment]
merged as 19abc86 thanks @shlomizadok

**Pull request closed without merging**